packages/core/src/web/app/actions/beambox/font-funcs.ts

```diff
@@ -682,51 +686,93 @@ const convertTextToPath = async (
 
       res =
         (await convertTextToPathByGhost(textElement, isFilled, font)) ||
-        convertTextToPathByFontkit(textElement, fontObj);
+        convertTextToPathByFontkit(textElement, fontObj, pathPerChar);
     } else {
       res =
-        convertTextToPathByFontkit(textElement, fontObj) ||
+        convertTextToPathByFontkit(textElement, fontObj, pathPerChar) ||
         (await convertTextToPathByGhost(textElement, isFilled, font));
     }
 
     if (res) {
       const { moveElement, transform } = res;
       let { d } = res;
 
-      if (weldingTexts) {
-        d = weldPath(d);
-      }
+      if (typeof d === 'string') {
+        if (weldingTexts) {
+          d = weldPath(d);
+        }
 
-      const path = document.createElementNS(svgedit.NS.SVG, 'path') as unknown as SVGPathElement;
+        const path = document.createElementNS(svgedit.NS.SVG, 'path') as unknown as SVGPathElement;
 
-      newPathElement = path;
+        newPathElement = path;
 
-      setAttributes(path, {
-        ...textAttr,
-        d,
-        id: svgCanvas.getNextId(),
-        'stroke-dasharray': 'none',
-        'stroke-opacity': '1',
-        transform,
-        'vector-effect': 'non-scaling-stroke',
-      });
+        setAttributes(path, {
+          ...textAttr,
+          d,
+          id: svgCanvas.getNextId(),
+          'stroke-dasharray': 'none',
+          'stroke-opacity': '1',
+          transform,
+          'vector-effect': 'non-scaling-stroke',
+        });
 
-      if (textAttr['stroke-width']) {
-        path.setAttribute('stroke-width', (+textAttr['stroke-width'] / 2).toString());
-      }
+        if (textAttr['stroke-width']) {
+          path.setAttribute('stroke-width', (+textAttr['stroke-width'] / 2).toString());
+        }
```

Comment:
nitpick:
Can add a function `createPathElement` to handle this in both case,
```
path.addEventListener('mouseover', svgCanvas.handleGenerateSensorArea);
path.addEventListener('mouseleave', svgCanvas.handleGenerateSensorArea);
```
Can only add for all paths but not group for `pathPerChar`